### PR TITLE
Improve pubkey DAO stub

### PR DESCRIPTION
**Pull request by ezr-ondrej**
Implements all DAO functions properly.
Implements pubkey generation.
Hides out the DAO stub structure, so it can't be manipulated from outside.

### CODE REVIEW

File: internal/dao/stubs/pubkey_dao.go [hunk 1]
```diff
@@ -27,15 +28,42 @@ func getPubkeyDao(ctx context.Context) (dao.PubkeyDao, error) {
 	return getPubkeyDaoStub(ctx)
 }
 
-func (mock *PubkeyDaoStub) Create(ctx context.Context, pk *models.Pubkey) error {
-	mock.store = append(mock.store, pk)
+func (stub *pubkeyDaoStub) Create(ctx context.Context, pk *models.Pubkey) error {
+	pk.ID = stub.lastId + 1
```
Comment (adiabramovitch) [inline internal/dao/stubs/pubkey_dao.go:32]: Why are we changing the pk id and didn't we do that before?
I guess we want the pk ids to match our store order?

File: internal/services/pubkey_service_test.go [hunk 2]
```diff
@@ -9,29 +9,19 @@ import (
...
-	}}
-}
-
 func TestListPubkeysHandler(t *testing.T) {
 	ctx := identity.WithIdentity(t, context.Background())
-	ctx = dao_stubs.WithPubkeyDao(ctx, buildPkStore())
+	ctx = stubs.WithPubkeyDao(ctx)
+	_, err := stubs.GeneratePubkey(ctx, nil)
+	assert.Nil(t, err, fmt.Sprintf("Error while generating a pubkey: %v", err))
+	_, err = stubs.GeneratePubkey(ctx, nil)
```
Comment (adiabramovitch) [inline internal/services/pubkey_service_test.go:23]: Why have you chosen to send nil and create pubkey this way?

Comment (ezr-ondrej) [reply internal/services/pubkey_service_test.go:23]: On the other hand not allowing the shorthand would simplify the return to just error, which would be neat :thinking:

Comment (adiabramovitch) [reply internal/services/pubkey_service_test.go]: I think it is cleaner to pass an "empty" pub key

Comment (ezr-ondrej) [reply internal/services/pubkey_service_test.go]: oki :+1: :)

**Review by adiabramovitch — APPROVED** [review]

**Pull request merged**